test/check-machines-disks

```diff
@@ -1408,23 +1408,24 @@ class TestMachinesDisks(VirtualMachinesCase):
         b.wait_in_text("#vm-subVmTest1-system-state", "Running")
         self.goToVmPage("subVmTest1")
 
+        b.wait_visible("#vm-subVmTest1-disks-vdc-device")
         b.wait_visible("#vm-subVmTest1-disks-vdc-action-kebab")
         b.click("#vm-subVmTest1-disks-vdc-action-kebab button")
-        b.wait_visible("#vm-subVmTest1-disks-vdc-device")
+        b.wait_visible("#delete-vm-subVmTest1-disks-vdc")
         b.click("#delete-vm-subVmTest1-disks-vdc")
         b.wait_visible(".pf-c-modal-box")
         b.wait_in_text(".pf-c-modal-box__body .pf-c-description-list", "subVmTest1")
         b.wait_in_text("#delete-resource-modal-target", "vdc")
         b.wait_in_text("#delete-resource-modal-file", vdc_path)
-        b.click(".pf-c-modal-box__footer button:contains(Remove)")
-        b.wait_visible(".pf-c-modal-box__footer button.pf-m-in-progress")
+        b.click("#delete-resource-modal-primary")
+        b.wait_not_present(".pf-c-modal-box")
         # When live-detaching disks the guest OS needs to cooperate so that we can
         # see the disk getting detached in the UI.
         # Wait until we see the login prompt before attempting operation that need
         # the OS for fully respond
         with b.wait_timeout(180):
             b.wait_not_present("#vm-subVmTest1-disks-vdc-device")
-        b.wait_not_present(".pf-c-modal-box")
+            wait(lambda: "vdc" not in m.execute("virsh domblklist subVmTest1"), delay=1)
```

Comment:
This is confusing. wait() is not a Browser class function, and thus not affected by `b.wait_timeout()`. Please move it outside of the context.

Also: Normally, it's the *delete* operation (i.e. clicking the button in the modal) which may be slow, not updating the UI after the dialog closed. So are you sure that the `b.wait_not_present(".pf-c-modal-box")` in line 1421 is fine with the default timeout?

The comment tries to justify the long timeout for updating the UI. But it doesn't match any more, as nothing actually waits for the login promt any more here. Can you please clean this up?

Reply:
Thanks for the update. I see you completely dropped the wait_timeout(), if that works, that's fine.